aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -359,6 +454,13 @@ static void copy_kernel_cuda(TensorIterator& iter, bool non_blocking) {
   int64_t nbytes = iter.numel() * iter.element_size(0);
   CUDAStream stream = getCurrentCUDAStream();
 
+  // Try optimized 1D/2D non-contiguous path first for both blocking and non-blocking
+  if (iter.ndim() <= 2 && !iter.is_contiguous()) {       
```

Comment:
I would really prefer not to send 1d case here, that's a lot of extra work to set sizes and pitch, and unclear if cudaMemcpy2DAsync doesn't have perf regressions compared to cudaMemcpyAsync